modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -0,0 +1,288 @@
+import * as _ from 'lodash';
+import * as nock from 'nock';
+import * as openpgp from 'openpgp';
+import * as should from 'should';
+import * as sinon from 'sinon';
+
+import { TestBitGo } from '@bitgo/sdk-test';
+import { BitGo } from '../../../../../src/bitgo';
+import {
+  common,
+  Keychain,
+  Wallet,
+  Ecdsa,
+  ECDSA,
+  EcdsaUtils,
+  ECDSAMethods,
+} from '@bitgo/sdk-core';
+
+const encryptNShare = ECDSAMethods.encryptNShare;
+type KeyShare = ECDSA.KeyShare;
+
+describe('TSS Utils:', async function () {
+  let sandbox: sinon.SinonSandbox;
+  let MPC: Ecdsa;
+  let bgUrl: string;
+  let tssUtils: EcdsaUtils;
+  let wallet: Wallet;
+  let bitgoKeyShare;
+  let userKeyShare: KeyShare;
+  let backupKeyShare: KeyShare;
+  let userGpgKey: openpgp.SerializedKeyPair<string> & {
+    revocationCertificate: string;
+  };
+
+  const coinName = 'gteth';
+
+  beforeEach(function () {
+    sandbox = sinon.createSandbox();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  before(async function () {
+    MPC = new Ecdsa();
+    const m = 2;
+    const n = 3;
+    userKeyShare = await MPC.keyShare(1, m, n);
+    backupKeyShare = await MPC.keyShare(2, m, n);
+    userGpgKey = await openpgp.generateKey({
+      userIDs: [
+        {
+          name: 'test',
+          email: 'test@test.com',
+        },
+      ],
+    });
+    bitgoKeyShare = await MPC.keyShare(3, m, n);
+
+    const bitGoGPGKey = await openpgp.generateKey({
+      userIDs: [
+        {
+          name: 'bitgo',
+          email: 'bitgo@test.com',
+        },
+      ],
+    });
+    const constants = {
+      mpc: {
+        bitgoPublicKey: bitGoGPGKey.publicKey,
+      },
+    };
+
+    nock('https://bitgo.fakeurl')
+      .persist()
+      .get('/api/v1/client/constants')
+      .reply(200, { ttl: 3600, constants });
+
+    const bitgo = TestBitGo.decorate(BitGo, { env: 'mock' });
+    bitgo.initializeTestVars();
+
+    const baseCoin = bitgo.coin(coinName);
+
+    bgUrl = common.Environments[bitgo.getEnv()].uri;
+
+    const walletData = {
+      id: '5b34252f1bf349930e34020a00000000',
+      coin: 'gteth',
+      coinSpecific: {},
+    };
+    wallet = new Wallet(bitgo, baseCoin, walletData);
+    tssUtils = new EcdsaUtils(bitgo, baseCoin, wallet);
+  });
+
+  describe('TSS key chains:', async function() {
+    it('should generate TSS key chains', async function () {
+      const nockedBitGoKeychain = await nockBitgoKeychain({
```

Comment:
The nockBitgoKeychain is implementing the same logic right for key generation?, do we need to reimplement it to test it by comparing?, i think its better to test after keygen, every property is coming in as expected and may be decrypt and make sure we are getting back the proper values?